### PR TITLE
Ziga/gateway metamask connect button

### PR DESCRIPTION
**Pull request by zkokelj**
### Why this change is needed

For the users to select which accounts they want to connect from metamask

### What changes were made as part of this PR

- new connect button and functionality
- small code refactoring
- added different testnet names

### PR checks pre-merging

Please indicate below by ticking the checkbox that you have read and performed the required
[PR checks](https://github.com/obscuronet/obscuro-internal/blob/main/dev-ops-docs/dev-pr-checks.md)

- [ ] PR checks reviewed and performed

### CODE REVIEW

File: tools/walletextension/api/staticOG/javascript.js [hunk 1]
```diff
@@ -156,6 +170,18 @@ async function connectAccount() {
...
 }
 
+async function isMetamaskConnected() {
+    let accounts;
+    try {
+        accounts = await provider.listAccounts()
+        return accounts.length > 0;
+
+    } catch (error) {
+        console.log("Unable to get accounts")
```
Comment (otherview) [inline tools/walletextension/api/staticOG/javascript.js:180]: should this return false ?

**Review by otherview — APPROVED** [review]
minor thing, non-blocker lgtm 👍

File: tools/walletextension/api/staticOG/javascript.js [hunk 2]
```diff
@@ -209,113 +236,123 @@ async function populateAccountsTable(document, tableBody, userID) {
...
+            // error 4902 means that the chain does not exist
+            if (switched === 4902 || !isValidUserIDFormat(await getUserID())) {
+                // join the network
+                const joinResp = await fetch(
+                    pathJoin, {
+                        method: methodGet,
+                        headers: jsonHeaders,
+                    });
+                if (!joinResp.ok) {
+                    return
```
Comment (otherview) [inline tools/walletextension/api/staticOG/javascript.js:315]: Should something be logged to the console and/or user here ?

**Review by otherview — APPROVED** [review]
Would had some info to the user or console if the join fails - but not a blocker imo, lgtm

File: tools/walletextension/api/staticOG/javascript.js (hunk 2 above)
Comment (zkokelj) [reply tools/walletextension/api/staticOG/javascript.js:315]: Added console log and message to user to try again later.

**Pull request merged**